src/mongocxx/test/spec/unified_tests/operations.cpp

```diff
@@ -1548,10 +1548,25 @@ document::value rewrap_many_datakey(entity::map& map,
     client_encryption& client_encryption = map.get_client_encryption(object);
     auto result = client_encryption.rewrap_many_datakey(filter.view(), rewrap_opts);
 
-    auto maybe_doc = result.result();
-    if (maybe_doc) {
-        auto doc =
-            make_document(kvp("result", make_document(kvp("bulkWriteResult", maybe_doc.value()))));
+    auto maybe_bulk_write = result.result();
+    if (maybe_bulk_write) {
+        auto bulk_write_result = maybe_bulk_write.value();
+        bsoncxx::builder::basic::document basic_builder{};
+        auto matched = bulk_write_result.matched_count();
+        auto modified = bulk_write_result.modified_count();
+
+        basic_builder.append(kvp("deletedCount", bulk_write_result.deleted_count()));
+        basic_builder.append(kvp("insertedCount", bulk_write_result.inserted_count()));
+        basic_builder.append(kvp("upsertedCount", bulk_write_result.upserted_count()));
+        basic_builder.append(kvp("upsertedIds", make_document()));
+        if (matched) {
+            basic_builder.append(kvp("matchedCount", matched));
+        }
+        if (modified) {
+            basic_builder.append(kvp("modifiedCount", modified));
+        }
+        auto doc = make_document(
+            kvp("result", make_document(kvp("bulkWriteResult", basic_builder.extract()))));
```

Comment:
Converting a bulk write result to a BSON document occurs elsewhere in this file. Consider adding a helper to convert bulk write result to a BSON document, which includes the `upsertedIds`:

```c++
document::value bulk_write_result_to_document(const mongocxx::result::bulk_write& result) {
    builder::basic::document upserted_ids_builder;
    for (auto&& index_and_id : result.upserted_ids()) {
        upserted_ids_builder.append(
            kvp(std::to_string(index_and_id.first), index_and_id.second.get_int32().value));
    }

    return make_document(kvp("matchedCount", result.matched_count()),
                         kvp("modifiedCount", result.matched_count()),
                         kvp("deletedCount", result.deleted_count()),
                         kvp("insertedCount", result.inserted_count()),
                         kvp("upsertedCount", result.upserted_count()),
                         kvp("upsertedIds", upserted_ids_builder.extract()));
}
```

The helper can be used here:

```suggestion
        auto doc = make_document(
            kvp("result",
                make_document(
                    kvp("bulkWriteResult", bulk_write_result_to_document(bulk_write_result)))));
        return doc;
```